### DISCUSSION TITLE
SWAR Config-Change and Restart Preserves Work-in-Progress

### DISCUSSION THREAD

**Discussion by Cminer3207** [swar/Swar-Chia-Plot-Manager#357] [General]
Restarting the SWAR Plot Manager DOES allow you to continue plotting existing Temp Plots. 

python manager.py restart 

HOWEVER!! 
You cannot change your drive configurations! 
(The plotter is actively working on those drives so it can’t stop).
If you change drives you need to shutdown, clear the Temp Directory, and start over again. 
You CAN change all the other parameters... (concurrency, stagger times, threads etc) 

You need to close your terminal window and wait for at least one minute before opening it and reopening the View (python manager.py view) 

If you don’t close, wait, reopen... it will look like the system has stalled... It hasn’t... It’s still running (assuming you didn’t change the drive configuration)

**Comment by Kali123411** [comment]
I thought you could change your destination drive if you copied the job to a new job then set the new location and zero out the numbers on the original job.

"Please note that if you change any of the directories for a job, it will mess with existing jobs and manager and view will not be able to identify the old job. If you are changing job directories while having active plots, please change the max_plots for the current job to 0 and make a separate job with the new directories. I do not recommend changing directories while plots are running."

**Reply by Cminer3207** [reply]
Yes. 
You are telling the manager to finish the Plot it is working on and not start anymore plots for this job order (because they will end up in the wrong place). 

All new plots will be created (in their new drive location) by the new job order and the first job order will end.

**Reply by Cminer3207** [reply]
If you do the 0-restart and you’re manager isn’t running at your Max_Concurrent level, it will immediately start Phase 1 plots (ignoring your Max_Phase1 limitations. 

You’ll basically trash your stagger until the OriginalJob is finished. (Not a big deal if you have extra CPU power to run extra Phase1’s) 

You should wait to retart until you have a plot that is in just about to finish Phase 1.
